### PR TITLE
Create validations and model specs

### PR DESCRIPTION
**Pull request by trace-kadenyi**
Add the following validations:

1. For the User model:
- [x] Name must not be blank.
- [x] PostsCounter must be an integer greater than or equal to zero.

2. For the Post model:
- [x] Title must not be blank.
- [x] Title must not exceed 250 characters.
- [x] CommentsCounter must be an integer greater than or equal to zero.
- [x] LikesCounter must be an integer greater than or equal to zero.
3. Add unit specs for all of the models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  it 'bio should be less than 180 characters' do
+    subject.bio = 'a' * 181
+    expect(subject).to_not be_valid
+  end
+
+  it 'bio should be more than 20 characters' do
+    subject.bio = 'a' * 19
+    expect(subject).to_not be_valid
```
Comment (brytebee) [inline spec/models/user_spec.rb]: - [x] **Required:** Kindly add a few positive tests that validate the presence of and not only what isn't permitted.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe Post, type: :model do
```
Comment (brytebee) [inline spec/models/post_spec.rb:3]: Same as above.

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  it 'is only valid with a user' do
+    comment.users = user
+    expect(comment).to be_valid
+  end
+
+  it 'is valid with a post' do
+    comment.posts = post
+    expect(comment).to be_valid
+  end
```
Comment (brytebee) [inline spec/models/comment_spec.rb]: - [x] **Optional:** You could also add negative tests here to validate what is not permitted.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @TracK92👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️


- Kindly add the procedure for running your tests in your README under the section `Run tests` as specified in the [Microverse README template.](https://github.com/microverseinc/readme-template)

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 3 above)
Comment (trace-kadenyi) [reply spec/models/comment_spec.rb]: Hello, 
I made all the changes, optional and required. Thanks for the suggestions.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED ✔️ 🟢 

## Hi @TracK92 👋, 

You have done a terrific job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] Please add the `tmp/cache/bootsnap` to your `.gitignore` as they are too large and you don't want them in you remote repo 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**